### DISCUSSION TITLE
Use with external NGINX possible ?

### DISCUSSION THREAD

**Discussion by michaelehlers** [llamafilm/tesla-http-proxy-addon#40] [Q&A]
Hi there,

I really appreciate thet things are moving. I am desperately waiting for a solution as everything I have done concerning smart charging does not work anymore and would love to get it working again.

My setup is that I have an 'external' reverse proxy running on a synology exposing my HA instance to www.

Is it than still possible to use this solution ? Or is it mandatory running nginx locally on the HA ?

Or maybe is there any thing that I could test ?

Thanks a lot to everbody working on that topic !

**Comment by llamafilm** [comment]
Some other people requested the same thing in #6 and I believe it should be possible.  You need to configure DNS on your proxy to reach this addon using its hostname (c03d64a7-tesla-http-proxy) on port 8099.  And host the public key somewhere.

**Reply by michaelehlers** [reply]
ahhhh...  thx a lot ! Didn't see that req, sorry ;)

Will follow on that one !

**Reply by 3lbart0** [reply]
I am exactly in the same situation as you, I have tried several proxy configurations, did not manage to make it work.
Did you manage to make the addon work with external NGINX ?

**Reply by bdeluca** [reply]
I kinda have an external NGINX, its not on talking to the addon through the internal docker network.  There are two peices you need to server the public key thats in /share/tesla and the port on 8099, I set up a http server to serve the /share/tesla files and then uses a location to point at it.
